File: software/examples/python/example_threshold.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-  

HOST = "localhost"
PORT = 4223
UID = "ABC" # Change to your UID

from tinkerforge.ip_connection import IPConnection
from tinkerforge.bricklet_voltage import Voltage

# Callback for voltage smaller than 5V
def cb_reached(voltage):
    print('Voltage dropped below 5V: ' + str(voltage/1000.0))

if __name__ == "__main__":
    ipcon = IPConnection() # Create IP connection
    v = Voltage(UID, ipcon) # Create device object

    ipcon.connect(HOST, PORT) # Connect to brickd
    # Don't use device before ipcon is connected

    # Get threshold callbacks with a debounce time of 10 seconds (10000ms)
    v.set_debounce_period(10000)

    # Register threshold reached callback to function cb_reached
    v.register_callback(v.CALLBACK_VOLTAGE_REACHED, cb_reached)

    # Configure threshold for "smaller than 5V" (unit is mV)
    v.set_voltage_callback_threshold('<', 5*1000, 0)

    raw_input('Press key to exit\n') # Use input() in Python 3
    ipcon.disconnect()

File: software/examples/javascript/ExampleThreshold.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'abd2'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var v = new Tinkerforge.BrickletVoltage(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        // Get threshold callbacks with a debounce time of 10 seconds (10000ms)
        v.setDebouncePeriod(10000);
        // Configure threshold for "smaller than 5V" (unit is mV)
        v.setVoltageCallbackThreshold('<', 5*1000, 0);
    }
);

// Register threshold reached callback
v.on(Tinkerforge.BrickletAnalogIn.CALLBACK_VOLTAGE_REACHED,
    // Callback for voltage smaller than 5V
    function(voltage) {
        console.log('Voltage: '+voltage/1000+' V');
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);
